b2sdk/api.py

```diff
@@ -312,18 +324,21 @@ def list_buckets(self, bucket_name=None, bucket_id=None):
         """
         # Give a useful warning if the current application key does not
         # allow access to the named bucket.
-        self.check_bucket_restrictions(bucket_name)
+        assert not (bucket_name and bucket_id)
+        if bucket_name:
```

Comment:
I cannot comment on existing lines...
in doc string there is:
```
you must specify the bucket name, or the request will be unauthorized.
```

and in api documentation:
```
When making a b2_list_buckets request with a key restricted to one bucket, you must include the bucket's ID (or name) in the request.
```